cloudinit/sources/helpers/vultr.py

```diff
@@ -0,0 +1,254 @@
+
+    return network
+
+
+# Input Metadata and generate public network config part
+def generate_public_network_interface(interfaces):
+    interface_name = get_interface_name(interfaces[0]['mac'])
+    if not interface_name:
+        raise RuntimeError(
+            "Interface: %s could not be found on the system" %
+            interfaces[0]['mac'])
+
+    netcfg = {
+        "name": interface_name,
+        "type": "physical",
+        "mac_address": interfaces[0]['mac'],
+        "accept-ra": 1,
+        "subnets": [
+            {
+                "type": "dhcp",
+                "control": "auto"
+            },
+            {
+                "type": "dhcp6",
+                "control": "auto"
+            },
+        ]
+    }
+
+    # Check for additional IP's
+    additional_count = len(interfaces[0]['ipv4']['additional'])
+    if "ipv4" in interfaces[0] and additional_count > 0:
+        for additional in interfaces[0]['ipv4']['additional']:
+            add = {
+                "type": "static",
+                "control": "auto",
+                "address": additional['address'],
+                "netmask": additional['netmask']
+            }
+            netcfg['subnets'].append(add)
+
+    # Check for additional IPv6's
+    additional_count = len(interfaces[0]['ipv6']['additional'])
+    if "ipv6" in interfaces[0] and additional_count > 0:
+        for additional in interfaces[0]['ipv6']['additional']:
+            add = {
+                "type": "static6",
+                "control": "auto",
+                "address": additional['address'],
+                "netmask": additional['netmask']
+            }
+            netcfg['subnets'].append(add)
+
+    # Add config to template
+    return netcfg
+
+
+# Input Metadata and generate private network config part
+def generate_private_network_interface(interfaces):
+    interface_name = get_interface_name(interfaces[1]['mac'])
+    if not interface_name:
+        raise RuntimeError(
+            "Interface: %s could not be found on the system" %
+            interfaces[1]['mac'])
+
+    netcfg = {
+        "name": interface_name,
+        "type": "physical",
+        "mac_address": interfaces[1]['mac'],
+        "accept-ra": 1,
+        "subnets": [
+            {
+                "type": "static",
+                "control": "auto",
+                "address": interfaces[1]['ipv4']['address'],
+                "netmask": interfaces[1]['ipv4']['netmask']
+            }
+        ]
+    }
+
+    return netcfg
+
+
+# Generate the vendor config
+# This configuration is to replicate how
+# images are deployed on Vultr before Cloud-Init
+def generate_config(md):
+    # Create vendor config
+    config_template = copy.deepcopy(md['vendor-config'])
+
+    # Add generated network parts
+    config_template['network'] = generate_network_config(md)
+
+    # Linux specific packages
+    if util.is_Linux():
+        config_template['packages'].append("ethtool")
```

Comment:
this is less than ideal.  the reason is the way that cloud-init merges user-data and vendor-data.
if the user provides 'packages' in their user-data (and many users *will* provide packages).. then they'll trump your list here :cry: .

I don't have a really good solution for this, though.

so just file this under the "remember this" bucket.

Reply:
I have several options I can go with here.

1. Install it inside the vendor script
2. Host it and pull down the individual binary for use
3. Base64 blob somewhere and then that in the config

Which do you feel would be the most appropriate solution here? I am hesitant to ignore another major requirement of ours, especially related to networking performance.

Reply:
@smoser Hate to bug you, but still need some input on how to best go about this.

Reply:
Eh sorry to bother you, we ended up going another route here for this.

Reply:
@eb3095 what did that mean?  Another route as in "drop this PR"? or just wrt startup script.

Reply:
> @eb3095 what did that mean? Another route as in "drop this PR"? or just wrt startup script.

No no no, just this specific portion. We ended up moving this to our end so we can better evaluate our options for now and removed it from the code. We will have another update to this soon.